apps/shipments/admin/shipment.py

```diff
@@ -151,6 +151,16 @@ def route_link(self, obj):
 class HistoricalShipmentAdmin(BaseModelHistory, ShipmentAdmin):
     readonly_fields = [field.name for field in Shipment._meta.get_fields()]
 
+    def get_fieldsets(self, request, obj=None):
+        """
+        Hook for specifying fieldsets.
+        """
+        list_fieldsets = list(self.fieldsets)
+        list_fieldsets[0][1]['fields'] = tuple(list(list_fieldsets[0][1]['fields'])[:-1] + [('contract_version',)])
+        self.fieldsets = tuple(list_fieldsets)
```

Comment:
This is a pretty inefficient way to go about this; tuple comprehension/filtering could be used to omit fields in a tuple, rather than converting back/forth between lists and tuples.

Reply:
ok, I can look into that. When I originally searched for a solution online most things suggested converting to/from a list.

Reply:
Due to the way the admin models are subclassed here, this actually overwrites the base Shipment admin fieldset and removes the route_link from the Shipment page.  As you rework this for efficiency, please also make sure this does not affect the Shipment admin page.